base/range.jl

```diff
@@ -920,68 +926,37 @@ end
 const OverflowSafe = Union{Bool,Int8,Int16,Int32,Int64,Int128,
                            UInt8,UInt16,UInt32,UInt64,UInt128}
 
-function getindex(v::UnitRange{T}, i::Integer) where {T<:OverflowSafe}
-    @inline
-    i isa Bool && throw(ArgumentError("invalid index: $i of type Bool"))
+function _getindex(v::UnitRange{T}, i::Integer) where {T<:OverflowSafe}
     val = v.start + (i - oneunit(i))
     @boundscheck _in_unit_range(v, val, i) || throw_boundserror(v, i)
     val % T
 end
 
-function getindex(v::OneTo{T}, i::Integer) where T
-    @inline
-    i isa Bool && throw(ArgumentError("invalid index: $i of type Bool"))
-    @boundscheck ((i > 0) & (i <= v.stop)) || throw_boundserror(v, i)
-    convert(T, i)
-end
-
-function getindex(v::AbstractRange{T}, i::Integer) where T
-    @inline
-    i isa Bool && throw(ArgumentError("invalid index: $i of type Bool"))
-    @boundscheck checkbounds(v, i)
-    convert(T, first(v) + (i - oneunit(i))*step_hp(v))
-end
-
+checkbounds(::Type{Bool}, v::AbstractRange, i::Integer) = ((i > 0) & (i <= v.stop))
```

Comment:
This definition was only valid for numeric ranges of type OneTo. It is not valid to assume all AbstractRange have a `stop` field.
```
LoadError(at "sysimg.jl" line 3: LoadError(at "Base.jl" line 292: LoadError(at "intfuncs.jl" line 788: MethodError(f=Base.isless, args=(1, Char(0x39000000)), world=0x00002e22))))
 ```

Reply:
Thanks for catching this oversight.
```suggestion
checkbounds(::Type{Bool}, v::AbstractRange, i::Integer) = (i > 0) & (i <= length(v))
```
Calling a function is better form anyhow

Reply:
With https://github.com/JuliaLang/julia/pull/50530, the generic fallback should be equally efficient
```suggestion
```